core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -4862,6 +4871,10 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
 #if WASM_ENABLE_GC != 0
             /* area of frame_ref */
             all_cell_num += (cell_num_of_local_stack + 3) / 4;
+            /* cells occupied by locals, POP_REF should not clear frame_ref for
+             * these cells */
+            local_cell_num =
+                cur_func->param_cell_num + cur_func->local_cell_num;
```

Comment:
Should also update `local_cell_num` when function return?

Reply:
Yes, updated
And should we also update `all_cell_num`?

Reply:
`all_cell_num` isn't used after allocating the frame, it should be unnecessary to restore it.

Reply:
Got it, thanks!